File: python/arrays_and_hashing/128_longest_consecutive_sequence.py
# https://leetcode.com/problems/longest-consecutive-sequence/

class Solution:
    def longestConsecutive(self, nums: list[int]) -> int:
        num_set = set(nums)
        longest = 0

        for n in nums:
            if (n - 1) in num_set:
                continue

            length = 1
            while (n + length) in num_set:
                length += 1

            longest = max(longest, length)

        return longest

File: go/arrays_and_hashing/128_longest_consecutive_sequence.go
package arraysandhashing

func LongestConsecutive(nums []int) int {
	set := make(map[int]bool, len(nums))
	for _, n := range nums {
		set[n] = true
	}

	longest := 0
	for _, n := range nums {
		if set[n-1] {
			continue
		}
		length := 1
		for {
			if !set[n+length] {
				break
			}
			length++
		}
		if length > longest {
			longest = length
		}
	}
	return longest
}
